tests/test_project_script.py

```
@@ -0,0 +1,160 @@
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+# Load the project script as a module
+_script_path = Path(__file__).parent.parent / "scripts" / "project"
+_spec = importlib.util.spec_from_loader("project_script", loader=None)
+_project_module = importlib.util.module_from_spec(_spec)
+
+# Read and execute the script to get the functions
+with open(_script_path) as f:
+    exec(f.read(), _project_module.__dict__)
```

Ruff (0.14.14)

[error] Use of `exec` detected

(S102)